javascript/vale-a-pena-aprender-vanillajs--com-certeza.md

```diff
@@ -0,0 +1,98 @@
+
+Então, existe alguma razão pela qual você deveria se importar em aprender Vanilla JS?
+
+É difícil se tornar um(a) bom(a) desenvolvedor(a) sem conhecer o principal da linguagem.
+
+Assim como você precisa entender os princípios fundamentais da lei antes de poder entrar em uma área específica da lei. Esta comparação é puramente acidental. 😉
+
+Eu entendo que a maioria dos aspirantes a desenvolvedor web querem ser contratados o quanto antes. Eu queria isso também.
+
+Parece muito mais fácil fazer um curso intensivo sobre o básico de JavaScript, procurar por um framework extravagante, desenvolver um app de ToDo List ([deixar um filhote morrer](https://medium.freecodecamp.org/every-time-you-build-a-to-do-list-app-a-puppy-dies-505b54637a5d)), enviar para o GitHub, e começar a procurar um emprego.
+
+## ...mas investir tempo em Vanilla JavaScript irá valer a pena a longo prazo
+
+Não me entenda mal. Eu não estou tendenciando contra frameworks JavaScript de maneira nenhuma. É examente o oposto. Muitos deles te ajudam a escrever código mais legível e de fácil manutenção. Eles te deixam escrever código mais abstrato que é mais fácil de debugar.
+
+Mas o ecossistema JavaScript está evoluindo muito rápido. Novos frameworks estão sendo criados. Novas funcionalidades estão sendo adicionadas as existentes. E o mais importante, muitos dos frameworks mais populares hoje serão substituídos, assim como Angular 1.
+
+Em tais circunstâncias, você ainda acha que o conhecimento de algum framework JavaScript em particular é o suficiente para um desenvolvedor web?
+
+Ou é melhor entender como a linguagem em que todas essas bibliotecas e frameworks são baseados funciona por debaixo dos panos?
+
+Sim, você está certo, é a segunda opção.
+
+Se você possui um sólido conhecimento dos fundamentos do JavaScript, a única coisa de que você precisa para se familiarizar ao começar a trabalhar com um novo framework é a *sintaxe* dele. As regras básicas serão as mesmas já que debaixo de todas as camadas de abstração, ainda é apenas JavaScript.
+
+Se seu conhecimento é limitado a um único framework JavaScript, você terá dificuldades em aprender um novo. Diferentes frameworks são normalmente baseados em um conjunto diferente de princípios do JavaScript. A longo prazo você irá passar mais tempo tentando entender vários frameworks e debugando o código que você escreveu neles.
+
+Todos os frameworks e bibliotecas JavaScripts são inevitavelmente baseados no Vanilla.
+
+Isto deve convencer qualquer um que o conhecimento de JavaScript puro é necessário a longo prazo. É um pré-requisito para qualquer desenvolvedor(a) de sucesso, especialmente para aqueles que trabalham desenvolvendo no ecossistema JavaScript.
+
+## Experiência Profissional
+
+Há não muito tempo atrás, eu olhei atrás para minha jornada de advogado a desenvolvedor web. Já faz 18 meses desde que escrevi minha primeira função JavaScript, e meu 10º mês trabalhando profissionalmente como um desenvolvedor front-end.
```

Comment:
Linkar `https://ideas.ataccama.com/i-stopped-being-a-lawyer-became-a-developer-and-its-awesome-5311e8d74882#.v3xurb9v5` em `jornada`.